microovn/api/types/services.go

```diff
@@ -1,13 +1,47 @@
 // Package types provides shared types and structs.
 package types
 
+import (
+	"fmt"
+)
+
 // Services - Slice with Service records.
 type Services []Service
 
 // Service  - A service.
 type Service struct {
-	// Service - name of Service
+	// Service - name of Service.
 	Service string `json:"service" yaml:"service"`
-	// Location - location of Service
+	// Location - location of Service.
 	Location string `json:"location" yaml:"location"`
 }
+
+// WarningSet - a set of warnings on the desired service state.
+type WarningSet struct {
+	// EvenCentral - are there an even number of central services which is
+	// inefficent due to how RAFT works.
+	EvenCentral bool `json:"EvenCentral" yaml:"EvenCentral"`
+	// FewCentral - are there not enough central services to handle one
+	// node failure.
+	FewCentral bool `json:"FewCentral" yaml:"FewCentral"`
+}
+
+// ServiceControlResponse (SCR) - a struct to return both a response and any
+// warnings, usually used when interfacing with the service control functions.
+type ServiceControlResponse struct {
+	// Message - any output needed from the service control functions.
+	Message string `json:"message" yaml:"message"`
+	// Warnings - the set of warnings with the desired state of services.
+	Warnings WarningSet `json:"warnings" yaml:"warnings"`
+}
+
+// PrettyPrint - Formats and prints contents of WarningSet object.
+func (w WarningSet) PrettyPrint() {
+	if w.EvenCentral {
+		fmt.Println("[central] Warning: Cluster with even number of members has same fault tolerance, but higher quorum requirements, than cluster with one less member.")
```

Comment:
I love the care you take to give our user a proper explanation about what is going on.

For a seasoned user, the messages do appear to be a bit long though, and they might be able to come to the conclusion of what is wrong sooner with a more terse message.

You'd be surprised how many people still operate in `80x24` terminals, and there it currently look like this:
![image](https://github.com/user-attachments/assets/42bbf4d8-1bc9-46a1-b7aa-64659f2b2fa1)

Would it make sense to keep this lovely explanation for documentation and for when the client has been invoked with the `--verbose` option, and provide a terser message when `--verbose` is not provided?
https://github.com/canonical/microovn/blob/803dc03ea847d214941330c74cc0ae0027550dc0/microovn/cmd/microovn/main.go#L20

I see @mkalcok pointed out that these should go to `STDERR`, I tend to agree.